src/validator.spec.ts

```diff
@@ -1,82 +1,220 @@
+      const result = validator({ value, customArgs: { date } });
 
-    // Assert
-    expect(result).toEqual({
-      succeeded: true,
-      message: '',
-      type: '',
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
     });
-  });
 
-  it('should return succeeded validation when it feeds value equals empty string', () => {
-    // Arrange
-    const value = '';
+    it('should return succeeded validation when value is a valid Date object with year earlier than customArgs date param', () => {
+      const value = new Date(2018);
+      const date = new Date(2019);
 
-    // Act
-    const result = validator({ value });
+      const result = validator({ value, customArgs: { date } });
 
-    // Assert
-    expect(result).toEqual({
-      succeeded: true,
-      message: '',
-      type: '',
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
     });
-  });
 
-  it('should overwrite default message when it feeds value and message', () => {
-    // Arrange
-    const value = 'test';
-    const message = 'other message';
+    it('should return succeeded validation when value is a valid Date object with year and month earlier than customArgs date param', () => {
+      const value = new Date(2018, 11);
+      const date = new Date(2018, 12);
 
-    // Act
-    const result = validator({ value, message });
+      const result = validator({ value, customArgs: { date } });
 
-    // Assert
-    expect(result).toEqual({
-      succeeded: false,
-      message: 'other message',
-      type: '',
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
+    });
+
+    it('should return succeeded validation when value is a valid Date object with year, month and days earlier than customArgs date param', () => {
+      const value = new Date(2018, 11, 24);
+      const date = new Date(2018, 11, 30);
+
+      const result = validator({ value, customArgs: { date } });
+
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
+    });
+
+    it('should return succeeded validation when value is a valid Date object with year, month, days and hours earlier than customArgs date param', () => {
+      const value = new Date(2018, 11, 30, 10);
+      const date = new Date(2018, 11, 30, 15);
+
+      const result = validator({ value, customArgs: { date } });
+
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
+    });
+
+    it('should return succeeded validation when value is a valid Date object with year, month, days, hours and minutes earlier than customArgs date param', () => {
+      const value = new Date(2018, 11, 24, 10, 15);
+      const date = new Date(2018, 12, 24, 10, 33);
```

Comment:
Please avoid dates with month `12`  in this format. It isn't `December`

Reply:
I think string format is better to get expected result:

```
const date = new Date('December 24, 2018 10:33:00');
```

Reply:
Changed to String format => f666b0294aa1bbded5fe45811d079f519857226f Line 137